ts/hooks/usePolling.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+
+export function usePolling<T>(fn: () => Promise<T>, delay: number) {
+  const [data, setData] = useState<T | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    const execute = async () => {
+      if (!isMounted) {
+        return;
+      }
+
+      try {
+        const result = await fn();
+        if (isMounted) {
+          setData(result);
+          setError(null);
+        }
+      } catch (err) {
+        window.log.warn('Polling error:', err);
```

Comment:
Anyway we can add the function name here so we have more context about the error?

Reply:
am not sure what you meant, by this, so I made something